src/common/row_operations/row_heap_scatter.cpp

```diff
@@ -7,6 +7,81 @@ namespace duckdb {
 
 using ValidityBytes = TemplatedValidityMask<uint8_t>;
 
+NestedValidity::NestedValidity(data_ptr_t validitymask_location)
+    : list_validity_location(validitymask_location), struct_validity_locations(nullptr), entry_idx(0), idx_in_entry(0) {
+}
+
+NestedValidity::NestedValidity(data_ptr_t *validitymask_locations, idx_t child_vector_index)
+    : list_validity_location(nullptr), struct_validity_locations(validitymask_locations), entry_idx(0),
+      idx_in_entry(0) {
+	ValidityBytes::GetEntryIndex(child_vector_index, entry_idx, idx_in_entry);
+}
+
+void NestedValidity::SetInvalid(idx_t idx) {
+	if (list_validity_location) {
+		// Is List
+		auto entry_offset = idx / 8;
+		auto bit_offset = idx % 8;
```

Comment:
Can we use this for computing the entry/bit offset?
```c++
idx_t entry_idx;
idx_t idx_in_entry;
ValidityBytes::GetEntryIndex(col_idx, entry_idx, idx_in_entry);
```

Reply:
Fixed!